flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/source/FileStoreSource.java

```diff
@@ -182,4 +216,65 @@ public FileStoreSourceSplitSerializer getSplitSerializer() {
     public PendingSplitsCheckpointSerializer getEnumeratorCheckpointSerializer() {
         return new PendingSplitsCheckpointSerializer(getSplitSerializer());
     }
+
+    private void writeObject(ObjectOutputStream out) throws IOException {
+        out.defaultWriteObject();
+        if (specifiedManifestEntries != null) {
+            new PartitionedDataManifestSerializer(
+                            fileStore.partitionType().getFieldCount(),
+                            fileStore.keyType(),
+                            fileStore.valueType())
+                    .serialize(specifiedManifestEntries, out);
+        }
+    }
+
+    private void readObject(ObjectInputStream in) throws ClassNotFoundException, IOException {
+        in.defaultReadObject();
+        if (in.available() > 0) {
+            specifiedManifestEntries =
+                    new PartitionedDataManifestSerializer(
+                                    fileStore.partitionType().getFieldCount(),
+                                    fileStore.keyType(),
+                                    fileStore.valueType())
+                            .deserialize(in);
+        }
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
We dont need to add `equals` and `hashCode` for this class.
The source is functional, not a pojo, and I don't think we need to implement these interfaces